lib/github_bitbucket_deployer/git.rb

```diff
@@ -59,12 +60,29 @@ def add_remote(remote = 'bitbucket')
       repo.add_remote(remote, bitbucket_repo_url)
     end
 
+    def repo_dir_path
+      @repo_dir_name ||= File.join(repo_dir, Zlib.crc32(git_repo_name).to_s)
+    end
+
+    def update_working_copy
+      logger.info('update_working_copy')
+      begin
+        exists_locally? ? pull : clone
+      rescue => e
+        if force_pristine_repo_dir && !@already_forced_pristine_repo_dir
+          make_repo_dir_pristine
+          retry
```

Comment:
This is the "magic". If the `force_pristine_repo_dir` is set and we haven't already "retried", we delete and recreate the repo-dir and then retry.